src/org/impact2585/frc2016/systems/AccelerometerSystem.java

```diff
@@ -36,17 +114,112 @@ public void init(Environment environ) {
 	 */
 	@Override
 	public void run() {
-		I2Cbus.read(DEVICE_ADDRESS, 14, buffer);
-		int j = 0;
-		//raw values are ax, ay, az, gx, gy, gz
-		for(int i = 0; i< rawValues.length; i++) {
-			if(j == 3)
-				j++;
-			rawValues[i] = (short) ((((short)buffer[j * 2]) << 8) | buffer[j * 2 + 1]);
-			j++;
+		//wait for the FIFO count to reach the dmp packet size
+		currentFIFOcount = getFIFOcount();
+		if (currentFIFOcount >= DMP_PACKET_SIZE) {
+			I2Cbus.read(0x74, DMP_PACKET_SIZE, FIFObuffer);
+			getQuaternionValues();
+			getGravity();
+			getAccelerationValues();
+			getGyroValues();
+		}
+	}
+	
+	/**
+	 * @returns the size of the FIFO buffer
+	 */
+	public short getFIFOcount() {
+		I2Cbus.read(0x72, 2, FIFOcount);
+		return (short)(((short)FIFOcount[1] << 8) | FIFOcount[0]);
+	}
+	
+	/**writes a bit to the slave register through I2C
+	 * @param register the register to write the bit to
+	 * @param position the bit to write to (0-7)
+	 * @param bit turn the bit on or off
+	 */
+	public void writeBit(int register, int position, boolean bit) {
+		byte b[] = new byte[8];
```

Comment:
Are you sure that this method actually writes to the I2C slave? It never calls `I2Cbus.write`. Also, please don't use one letter variable names.